src/types/observablemap.ts

```diff
@@ -279,10 +280,11 @@ export class ObservableMap<K = any, V = any>
     entries(): IterableIterator<IMapEntry<K, V>> {
         const self = this
         let nextIndex = 0
+        const keys = Array.from(this.keys())
```

Comment:
use iteratorToArray as well here for consistency?

Reply:
I think we can actually avoid the array here and reuse the iterator returned from keys().
However I first want it to match v5 as close as possible...
And I also want to take a look at how native Map handles modification of keys during iteration.